src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseGradingService.java

```diff
@@ -88,98 +87,83 @@ public ProgrammingExerciseGradingService(ProgrammingExerciseTestCaseService test
+     * @param semiAutomaticResult The latest manual result for the same submission
+     * @param newAutomaticResult The new automatic result
      * @param programmingExercise The programming exercise
-     * @return A new manual result
+     * @return The updated semi-automatic result
      */
-    private Result getNewManualResultFromResults(Result automaticResult, Result manualResult, ProgrammingExercise programmingExercise) {
-
-        // create a new manual result for the same submission as the automatic result
-        Result newResult = programmingSubmissionService.saveNewEmptyResult(automaticResult.getSubmission());
-
-        newResult.setAssessor(manualResult.getAssessor());
-        newResult.setAssessmentType(AssessmentType.SEMI_AUTOMATIC);
+    private Result updateSemiAutomaticResult(Result semiAutomaticResult, Result newAutomaticResult, ProgrammingExercise programmingExercise) {
         // this makes it the most recent result, but optionally keeps the draft state of an unfinished manual result
-        newResult.setCompletionDate(manualResult.getCompletionDate() != null ? automaticResult.getCompletionDate().plusSeconds(1) : null);
-        newResult.setHasFeedback(manualResult.getHasFeedback());
-        newResult.setRated(manualResult.isRated());
+        semiAutomaticResult.setCompletionDate(semiAutomaticResult.getCompletionDate() != null ? newAutomaticResult.getCompletionDate().plusSeconds(1) : null);
+
+        // remove old automatic feedback
+        semiAutomaticResult.getFeedbacks().removeIf(feedback -> feedback != null && feedback.getType() == FeedbackType.AUTOMATIC);
 
         // copy all feedback from the automatic result
-        for (Feedback feedback : automaticResult.getFeedbacks()) {
+        for (Feedback feedback : newAutomaticResult.getFeedbacks()) {
             Feedback newFeedback = feedback.copyFeedback();
-            newResult.addFeedback(newFeedback);
+            semiAutomaticResult.addFeedback(newFeedback);
         }
 
-        // copy only the non-automatic feedback from the manual result
-        for (Feedback feedback : manualResult.getFeedbacks()) {
-            if (feedback != null && feedback.getType() != FeedbackType.AUTOMATIC) {
-                Feedback newFeedback = feedback.copyFeedback();
-                newResult.addFeedback(newFeedback);
-            }
-        }
-
-        String resultString = updateManualResultString(automaticResult.getResultString(), newResult, programmingExercise);
-        newResult.setResultString(resultString);
-
-        // workaround to prevent that result.submission suddenly turns into a proxy and cannot be used any more later after returning this method
-        Submission tmpSubmission = newResult.getSubmission();
-        newResult = resultRepository.save(newResult);
-        newResult.setSubmission(tmpSubmission);
+        String resultString = updateManualResultString(newAutomaticResult.getResultString(), semiAutomaticResult, programmingExercise);
```

Comment:
are you sure that the order of parameters is correct here? 
Because in the old code it was updateManualResultString(automaticResult, newResult, progEx)
now it seems the other way round:
updateManualResultString(newAutomaticResult, semiAutomaticResult, progEx)

Reply:
No the order didn't change, the current method is: ` updateManualResultString(String resultString, Result result, ProgrammingExercise exercise)`